### PR TITLE
Updates to documentation

### PR DESCRIPTION
**Pull request by igomez06**
**Framework updates**

- Cluster provisioning README with config examples of our different clusters.
- Updates to documentation. Added more detail so our go docs can provide more help for our end users.

### CODE REVIEW

**Review by paynejacob — APPROVED** [review]

File: tests/v2/validation/provisioning/README.md [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+    "rootPass": "",
+    "sshPort": "22",
+    "sshUser": "",
+    "stackscript": "",
+    "stackscriptData": "",
+    "swapSize": "512",
+    "tags": "",
+    "uaPrefix": ""
+  },
+  "azureMachineConfig": {
```
Comment (izaac) [inline tests/v2/validation/provisioning/README.md:117]: Could we split these big json blocks to their respective cloud provider sub title?

Like: `## Azure example ##` then the json and the next title `## EKS example ##` (or a deeper subtitle level).

I've seen the doc in view document and that would help its readability

**Review by izaac — CHANGES_REQUESTED** [review]

File: tests/v2/validation/provisioning/README.md (hunk 1 above)
Comment (igomez06) [reply tests/v2/validation/provisioning/README.md]: Does that look better @izaac ?

Comment (izaac) [reply tests/v2/validation/provisioning/README.md]: Way better! thank you!

**Review by izaac — DISMISSED** [review]
LGTM

**Review by caliskanugur — DISMISSED** [review]

File: tests/framework/pkg/wait/watch.go [hunk 2]
```diff
@@ -6,9 +6,23 @@ import (
...
+// WatchCheckFunc is the function type of `check` needed for WatchWait ex)
+//  checkFunc := func(event watch.Event) (ready bool, err error) {
+// 		cluster := event.Object.(*apisV1.Cluster)
+// 		ready = cluster.Status.Ready
+// 		return ready, nil
+//  }
 type WatchCheckFunc func(watch.Event) (bool, error)
 
 // WatchWait uses the `watchInterface`  to wait until the `check` function to returns true.
+// ex) WatchWait for provisioning a cluster
```
[the remark this answers is not in the document]
Comment (igomez06) [reply tests/framework/pkg/wait/watch.go:18]: Usually you would have a whole new example file, but for this I found it to be too much. I don't think there is convention to use either `ex)` or `For Example` which do you prefer?

[the remark this answers is not in the document]
Comment (izaac) [reply tests/framework/pkg/wait/watch.go:18]: I don't personally have a preference but i would be leaned to whichever looks clearer when read from the generated doc

[the remark this answers is not in the document]
Comment (igomez06) [reply tests/framework/pkg/wait/watch.go]: Made the updates we discussed

**Comment by caliskanugur** [comment]
LGTM

**Review by izaac — DISMISSED** [review]
LGTM

Formatting looks good when running godoc from local

**Review by caliskanugur — APPROVED** [review]

**Review by izaac — APPROVED** [review]
LGTM

**Pull request merged**